### PR TITLE
Linked In Backend Integration Sign in only

### PR DESCRIPTION
**Pull request by MrMaksimize**

### CODE REVIEW

File: cloud/modules/linkedin/linkedin.js [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+      failureCallback(res);
+    }
+  });
+}
+
+exports.getParsePasswordFromID = function(profileID, salt) {
+  return crypto.createHash('md5').update(profileID + 'LiAccess' + salt).digest('hex');
+}
+
+exports.executeRequest = function(request_path, callbackFunction) {
```
Comment (glesage) [inline cloud/modules/linkedin/linkedin.js:138]: There is a nice way to do Get Requests if you want to use it.
Here's an example from my tinyUrl rest api stuff:

```
Parse.Cloud.httpRequest({
    url: 'http://tiny.cc/',
    params: {
      c : 'rest_api',
      format : 'json',
      longUrl : $parseFile.url,
      login  : 'resutext',
      apiKey : 'ecc1a578-f5b7-4aaa-beb1-d1510ed008ee'
    },
    success:
    failure:
```

File: cloud/modules/linkedin/linkedin.js [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+APICalls['companySearch'] = 'company-search:(companies,facets)?facet=location,us:84';
+
+// JOBS APIS
+// Be sure to change the JobId or facets accordingly
+APICalls['myJobSuggestions'] = 'people/~/suggestions/job-suggestions';
+APICalls['myJobBookmarks'] = 'people/~/job-bookmarks';
+APICalls['jobDetails'] = 'jobs/1452577:(id,company:(name),position:(title))';
+APICalls['jobSearch'] = 'job-search:(jobs,facets)?facet=location,us:84';
+
+exports.initialize2 = function(req, res) {
```
Comment (glesage) [inline cloud/modules/linkedin/linkedin.js:204]: So why have 2 initialize functions?

File: cloud/modules/linkedin/linkedin.js [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+    req.headers.cookie && req.headers.cookie.split(';').forEach(function( cookie ) {
+      var parts = cookie.split('=');
+      cookies[ parts[ 0 ].trim() ] = ( parts[ 1 ] || '' ).trim();
+    });
+
+    // If we have the access_token in the cookie skip the Oauth Dance and go straight to Step 3
+    if (cookies['LIAccess_token']){
+      // STEP 3 - Get LinkedIn API Data
+      // console.log("we have the cookie value" + cookies['LIAccess_token']);
+      //OauthStep3(req, response, cookies['LIAccess_token'], APICalls['peopleSearchWithKeywords']);
```
Comment (glesage) [inline cloud/modules/linkedin/linkedin.js:220]: So whats all this about... why all commented out?

File: cloud/resutext-user.js [hunk 4]
```diff
@@ -0,0 +1,103 @@
+
```
Comment (glesage) [inline cloud/resutext-user.js:1]: We'll prolly merge this with some of the other user files...
Or at least figure out which user handling should be in which files cus as of now we have 3 separate files for user handling...

File: cloud/resutext-user.js [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+  return promise;
+}
+
+
+exports.loginWithLinkedIn = function(req, res) {
+
+  console.log('initialize');
+
+  linkedInClient.initialize({
+    APIKey: 'v0wh3ponihe9',
```
Comment (glesage) [inline cloud/resutext-user.js:59]: What about putting all this API static stuff in globals.js?

File: cloud/resutext-user.js [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+    APIKeySecret: 'UmYOhdOAg8aS7dQI',
+    callbackURL: 'http://resutext.parseapp.com/auth',
+    redirectPostAuth: 'http://resutext.parseapp.com',
+    APIScope: 'r_basicprofile r_fullprofile r_emailaddress r_network r_contactinfo rw_nus rw_groups w_messages'
+    //accessToken: '' // Access token can be pulled from DB?
+  });
+
+  linkedInClient.authenticate(req, res, function(response) {
+    console.log('success on authenticate');
+    console.log('yay');
```
Comment (glesage) [inline cloud/resutext-user.js:69]: You mean yayo?

:P

File: cloud/resutext-user.js [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+    linkedInClient.getCurrentUserProfile(function(profileResponse) {
+      var clientSettings = linkedInClient.getCurrentSettings();
+      var userQuery = new Parse.Query(Parse.User);
```
Comment (glesage) [inline cloud/resutext-user.js:72]: I think we'll talk about it on friday but what about putting this in its own function? Or even making an abstract one and putting it in user-handler.js.. cus I'm doing similar searches several times based on different critierias...

File: cloud/modules/linkedin/linkedin.js (hunk 1 above)
Comment (MrMaksimize) [inline cloud/modules/linkedin/linkedin.js:138]: Oh I know i'm using it right below.

The reason I have to use &format=json is cuz their api sucks and can't detect json from a header, so it requires this query string

Comment (glesage) [inline cloud/modules/linkedin/linkedin.js:138]: oh... wut about &oauth2_access_token=' + settings.accessToken;

**Pull request merged**